packages/gestalt/src/SheetMobile.js

```diff
@@ -0,0 +1,183 @@
+// @flow strict
+import { type ElementConfig, type Node } from 'react';
+import Button from './Button.js';
+import Link from './Link.js';
+import AnimationProvider from './animation/AnimationContext.js';
+import DismissingElement from './animation/DismissingElement.js';
+import { useDeviceType } from './contexts/DeviceTypeProvider.js';
+import FullPage from './SheetMobile/FullPage.js';
+import PartialPage from './SheetMobile/PartialPage.js';
+
+type OnClickType = ({|
+  event:
+    | SyntheticMouseEvent<HTMLButtonElement>
+    | SyntheticKeyboardEvent<HTMLButtonElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  onDismissStart: () => void,
+|}) => void;
+
+type Props = {|
+  /**
+   * String that clients such as VoiceOver will read to describe SheetMobile when opened. See [Accessibility section](https://gestalt.pinterest.systems/web/sheetmobile#Accessibility) for more info.
+   */
+  accessibilityLabel?: string,
+  /**
+   * Specify the alignment of `heading` & `subHeading` strings. See the [Header variant](https://gestalt.pinterest.systems/web/sheetmobile#Heading) for more info.
+   */
+  align?: 'start' | 'center',
+  /**
+   * Adds a "back-arrow" IconButton for user interaction at the start of the header section. See the [header variant, back and forward navigation case](https://gestalt.pinterest.systems/web/sheetmobile#Header) for more info.
+   */
+  backIconButton?: {| accessibilityLabel: string, onClick: OnClickType |},
```

Comment:
good point

Reply:
> Does this accessibilityLabel need to be required? Can we use a default label here?

I don't think defaults might work well, because it depends on every case. Previous page? Previous step? I can discuss with Kate and figure out if there's a generic case

Reply:
I'm still seeing `OnClickType` instead of the actual type.

And naming in subcomponents can be addressed if/when needed — the naming for this component should match its actual mobile-only usage. (Also, TapArea uses `onTap`, not `onClick`.)